### PR TITLE
Missed native props

### PR DESCRIPTION
**Pull request by magrinj**

### CODE REVIEW

**Comment by cybergrind** [comment]
@isnifer could you please review this

File: src/components/PaymentCardTextField.ios.js [hunk 1]
```diff
@@ -135,8 +154,7 @@ export default class PaymentCardTextField extends Component {
           numberPlaceholder={numberPlaceholder}
           expirationPlaceholder={expirationPlaceholder}
           cvcPlaceholder={cvcPlaceholder}
-          onChange={this.handleChange}
-        />
+          onChange={this.handleChange}/>
```
Comment (isnifer) [inline src/components/PaymentCardTextField.ios.js]: revert this change

File: src/components/PaymentCardTextField.android.js [hunk 2]
```diff
@@ -113,6 +121,9 @@ export default class PaymentCardTextField extends Component {
           fontStyle={fontStyle}
           fontSize={fontSize}
           enabled={!disabled}
+          cardNumber={ card.number || null }
```
Comment (isnifer) [inline src/components/PaymentCardTextField.android.js]: Above in render:
```js
const cardNumber = card.number || null
// ...
```

As property:
```js
cardNumber={cardNumber}
```

File: src/components/PaymentCardTextField.android.js [hunk 3]
```diff
@@ -113,6 +121,9 @@ export default class PaymentCardTextField extends Component {
...
+          expDate={ card.expMonth || card.expYear ? `${card.expMonth}/${card.expYear}` : null }
```
Comment (isnifer) [inline src/components/PaymentCardTextField.android.js]: Above in render:
```js
const { expMonth = 'MM', expYear = 'YY' } = card
const expDate = expMonth || expYear ? `${expMonth}/${expYear}` : null
// ...
```

As property:
```js
expDate={expDate}
```

File: src/components/PaymentCardTextField.android.js [hunk 4]
```diff
@@ -113,6 +121,9 @@ export default class PaymentCardTextField extends Component {
...
+          securityCode={ card.cvc || null }
```
Comment (isnifer) [inline src/components/PaymentCardTextField.android.js]: Above in render:
```js
const securityCode = card.cvc || null
```

As property:
```js
securityCode ={securityCode}
```

File: src/components/PaymentCardTextField.android.js [hunk 5]
```diff
@@ -15,15 +15,23 @@ export default class PaymentCardTextField extends Component {
     ...View.propTypes,
     style: StyleSheetPropType(FieldStylePropType), // eslint-disable-line new-cap
 
-    setEnabled: PropTypes.bool,
+    // Android only
```
Comment (isnifer) [inline src/components/PaymentCardTextField.android.js]: remove this comment

**Comment by magrinj** [comment]
@isnifer I apply your changes, hope it's ok for you :)

File: src/components/PaymentCardTextField.android.js [hunk 6]
```diff
@@ -93,6 +100,9 @@ export default class PaymentCardTextField extends Component {
       color,
       ...fieldStyles
     } = StyleSheet.flatten(style)
+    const cardNumber = card.number || null;
```
Comment (isnifer) [inline src/components/PaymentCardTextField.android.js]: as you can see - we don't use semicolons

**Comment by isnifer** [comment]
@magrinj please, tell me. I see `card` object takes a boolean property called `valid` what it does?

**Comment by magrinj** [comment]
@isnifer Yes you're right, at this time it does nothing I'm going to remove it :)

File: src/components/PaymentCardTextField.android.js [hunk 7]
```diff
@@ -93,6 +100,9 @@ export default class PaymentCardTextField extends Component {
...
+    const expDate = (card.expMonth || card.expYear ? `${card.expMonth}/${card.expYear}` : null);
```
Comment (isnifer) [inline src/components/PaymentCardTextField.android.js]: remove braces

**Comment by magrinj** [comment]
@isnifer I remove braces and unused prop

File: README.md [hunk 8]
```diff
@@ -545,13 +545,18 @@ A text field component specialized for collecting credit/debit card information.
...
+* `placeholderColor` _String_ (IOS only) - The text placeholder color used in each child field.
 * `numberPlaceholder` _String_ - The placeholder for the card number field.
 * `expirationPlaceholder` _String_ - The placeholder for the expiration field.
 * `cvcPlaceholder` _String_ - The placeholder for the cvc field.
-* `disabled` _Bool_ (IOS only) - Enable/disable selecting or editing the field. Useful when submitting card details to Stripe.
-* `enabled` _Bool_ (Android only) - Enable/disable selecting or editing the field. Useful when submitting card details to Stripe.
+* `disabled` _Bool_ - Enable/disable selecting or editing the field. Useful when submitting card details to Stripe.
 * `onChange` _Func_ - This function will be called each input change.
+* `card` _Object_ - Accept a card object for the default value
+  * `valid` _Bool_ - Whether or not the form currently contains a valid card number, expiration date, and CVC.
```
Comment (isnifer) [inline README.md]: remove this info from readme

**Comment by magrinj** [comment]
I remove `valid` prop from README :)

**Comment by isnifer** [comment]
@magrinj please, also remove all semicolons

**Comment by magrinj** [comment]
Done :)

**Comment by magrinj** [comment]
Any news on this ? :)

**Comment by itsmepetrov** [comment]
@magrinj sorry for late response, I will try to review this PR before the end of the week.

File: src/components/PaymentCardTextField.android.js [hunk 9]
```diff
@@ -15,15 +15,21 @@ export default class PaymentCardTextField extends Component {
...
     style: StyleSheetPropType(FieldStylePropType), // eslint-disable-line new-cap
 
-    setEnabled: PropTypes.bool,
     backgroundColor: PropTypes.string,
-    cardNumber: PropTypes.string,
-    expDate: PropTypes.string,
-    securityCode: PropTypes.string,
+
+    card: PropTypes.shape({
+      number: PropTypes.string,
```
Comment (isnifer) [inline src/components/PaymentCardTextField.android.js]: Why does this field is a string?

File: src/components/PaymentCardTextField.android.js [hunk 10]
```diff
@@ -15,15 +15,21 @@ export default class PaymentCardTextField extends Component {
...
+      expMonth: PropTypes.number,
+      expYear: PropTypes.number,
+      cvc: PropTypes.string,
```
Comment (isnifer) [inline src/components/PaymentCardTextField.android.js]: And this too?

**Comment by isnifer** [comment]
@magrinj I tried to run this code on both platforms and what I've got:
### iOS
image

### Android
image

**Comment by magrinj** [comment]
I don't really understand what's happen, on android the CVC is not set ?

**Comment by magrinj** [comment]
Ok so I take a look for the android cvc problem, it looks like it's a problem with the library, in android log I got something like that when the input is set:

```
08-11 17:41:40.274 6924-6924/com.example D/TAG: setSecurityCode: 123
08-11 17:41:40.275 6924-6924/com.example D/CustomCardInputReactManager: onTextChanged: CCV = 123
08-11 17:41:40.275 6924-6924/com.example D/CustomCardInputReactManager: onTextChanged: CCV = 
08-11 17:41:40.276 6924-6924/com.example D/CustomCardInputReactManager: onTextChanged: CCV = 
08-11 17:41:40.277 6924-6924/com.example D/CustomCardInputReactManager: onTextChanged: CCV = 
08-11 17:41:40.277 6924-6924/com.example D/CustomCardInputReactManager: onTextChanged: CCV = 
08-11 17:41:40.278 6924-6924/com.example D/CustomCardInputReactManager: onTextChanged: CCV = 
08-11 17:41:40.278 6924-6924/com.example D/CustomCardInputReactManager: onTextChanged: CCV = 
08-11 17:41:40.279 6924-6924/com.example D/CustomCardInputReactManager: onTextChanged: CCV = 
```

So I don't really understand why it's happening, I try to force `currentCCV` in android code manually, but the CVC is still empty, don't really understand why :/

**Comment by isnifer** [comment]
@magrinj this problem in the Stripe or in the tipsi-stripe?

**Comment by magrinj** [comment]
@isnifer I think it's in CreditCardForm, I don't find any problem on tips-stripe side at the moment 😕

**Comment by magrinj** [comment]
Any news on this ? :)

**Comment by fbartho** [comment]
Nothing has moved on this PR in 2 years. There have been unfortunately many, conflicting changes on the `experimental` branch for the 8.0 release & deadline. I'm going to close this PR @magrinj for now -- but I would welcome a new PR based on there if this is still useful to you!

We're looking for a follow-on release in a month, so it wouldn't sit forever!
